chips/sam4l/src/usart.rs

```diff
@@ -773,11 +818,13 @@ impl hil::spi::SpiMaster for USART {
 
     /// Pass in a None to use the HW chip select pin on the USART (RTS).
     fn specify_chip_select(&self, cs: Self::ChipSelect) {
+        self.enable_clock();
         self.spi_chip_select.set(cs);
     }
 
     /// Returns the actual rate set
     fn set_rate(&self, rate: u32) -> u32 {
+        self.enable_clock();
```

Comment:
I'm worried that all these set/get functions could leave us with our clock enabled for an undefined period of time.